### PR TITLE
Implement use of multiple rpc targets in mbus

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR
This follows the pattern from the java implementation.

### CODE REVIEW

File: messagebus/src/vespa/messagebus/network/rpctargetpool.cpp [hunk 1]
```diff
@@ -4,23 +4,41 @@
...
+    RPCTarget::SP target =  _targets[_next++];
+    if ( ! target->isValid()) return RPCTarget::SP();
+    _lastUse = now;
+    return target;
+}
+
+bool
+RPCTargetPool::Entry::inUse(const LockGuard &) const {
+    for (const auto & target : _targets) {
+        if (target.use_count() > 1) return true;
```
Comment (geirst) [inline messagebus/src/vespa/messagebus/network/rpctargetpool.cpp:27]: Should this be "_> 0_" instead ?

File: messagebus/src/vespa/messagebus/network/rpctargetpool.cpp [hunk 2]
```diff
@@ -4,23 +4,41 @@
...
-    _targets(),
-    _timer(new SteadyTimer()),
-    _expireMillis(static_cast<uint64_t>(expireSecs * 1000))
+RPCTarget::SP
+RPCTargetPool::Entry::getTarget(const LockGuard &, uint64_t now) {
+    if (_next >= _targets.size()) {
+        _next = 0;
+    }
+    RPCTarget::SP target =  _targets[_next++];
+    if ( ! target->isValid()) return RPCTarget::SP();
```
Comment (geirst) [inline messagebus/src/vespa/messagebus/network/rpctargetpool.cpp:19]: Consider using braces.

File: messagebus/src/vespa/messagebus/network/rpctargetpool.cpp (hunk 1 above)
Comment (geirst) [reply messagebus/src/vespa/messagebus/network/rpctargetpool.cpp:27]: Consider using braces.

**Review by geirst — COMMENTED** [review]
Consider adding a basic unit test for RPCTargetPool (in targetpool.cpp) that verifies we get different targets for the same spec/address.

File: messagebus/src/vespa/messagebus/network/rpctargetpool.cpp (hunk 1 above)
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/network/rpctargetpool.cpp:27]: Braces done.
target is a shared_ptr so use_count will never be 0. If it is larger than 1 it means there is another reference than the one in the pool active.

File: messagebus/src/vespa/messagebus/network/rpctargetpool.cpp (hunk 2 above)
Comment (baldersheim) [reply messagebus/src/vespa/messagebus/network/rpctargetpool.cpp:19]: Fixed.

**Review by geirst — COMMENTED** [review]
👍  We should also add a unit test for this.

**Pull request merged**